okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/ThreadInterruptTest.java

```diff
@@ -133,4 +184,103 @@ private void interruptLater(final int delayMillis) {
     };
     interruptingCow.start();
   }
+
+  private static class FakeNetwork implements Network {
+    private final InetAddress[] addresses;
+
+    public FakeNetwork(InetAddress[] addresses) {
+      this.addresses = addresses;
+    }
+
+    @Override public InetAddress[] resolveInetAddresses(String host) throws UnknownHostException {
+      return addresses;
+    }
+  }
+
+  private static InetAddress getNonLoopbackLocalAddress() throws SocketException {
+    InetAddress localAddress = null;
+    Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+    outer: while (interfaces.hasMoreElements()) {
+      NetworkInterface networkInterface = interfaces.nextElement();
+      if (networkInterface.isUp() && !networkInterface.isLoopback()) {
+        Enumeration<InetAddress> addresses = networkInterface.getInetAddresses();
+        while (addresses.hasMoreElements()) {
+          localAddress = addresses.nextElement();
+          break outer;
+        }
+      }
+    }
+    if (localAddress == null) {
+      fail("No network");
+    }
+    return localAddress;
+  }
+
+  private static InetAddress getBlackholeAddress(InetAddress localAddress)
+      throws UnknownHostException {
+    byte[] addressBytes =
+        localAddress instanceof Inet4Address ? IPV4_BLACKHOLE_ADDRESS : IPV6_BLACKHOLE_ADDRESS;
+    return InetAddress.getByAddress(addressBytes);
+  }
+
+  private static class TimeoutCountingSocketFactory extends DelegatingSocketFactory {
```

Comment:
I think you'll want to change DelegatingSocketFactory so that configureSocket returns a Socket. That way we define these 5 overloads only once!